core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -673,7 +673,8 @@ private boolean deleteFromAllStoresSync(Object key, int segment, Predicate<? sup
             semaphore.acquire();
             return advancedCacheLoader.entryPublisher(filter, fetchValue, fetchMetadata);
          }, Semaphore::release)
-               .subscribeOn(persistenceScheduler);
+               .subscribeOn(persistenceScheduler)
+               .observeOn(cpuScheduler);
```

Comment:
Okay. does look better

Reply:
Also another unexpected benefit of using `observeOn` is that the semaphore is released much quicker as it only needs to consume the original store Publisher before it is released. Originally the last `onNext` and `onComplete` also had to finish before releasing it before. This is what made the stop test so annoying to fix.

Reply:
Might be misunderstand myself, but there is no `=` in this case. And it sounds somewhat familiar to what this would look like if there was a variable assignment.

Reply:
Sorry, I noticed that in several places the chained methods were indented at 6 spaces instead of being aligned with the first method in the chain and I assumed this was the same issue. Splitting after `=` is only relevant when chained method calls are aligned, because without splitting the first call is way to the right.

I finally tried this exact case and I realized how bad IntelliJ really is here, but I would have probably solved it another way:

```java
         return Flowable.using(publisherSemaphoreCallable,
                               semaphore -> {
                                  semaphore.acquire();
                                  return advancedCacheLoader.entryPublisher(filter, fetchValue, fetchMetadata);
                               }, Semaphore::release)
                        .subscribeOn(persistenceScheduler)
                        .observeOn(cpuScheduler);
```